priam/src/main/java/com/netflix/priam/services/BackupTTLService.java

```diff
@@ -130,15 +126,45 @@ public void execute() throws Exception {
             // Walk over the file system iterator and if not in map, it is eligible for delete.
             new MetaFileWalker().readMeta(localFile);
 
-            if (logger.isDebugEnabled())
-                logger.debug("Files in meta file: {}", filesInMeta.keySet().toString());
+            logger.info("No. of component files loaded from meta file: {}", filesInMeta.size());
 
             // Delete the meta file downloaded locally
             FileUtils.deleteQuietly(localFile.toFile());
 
+            // If there are no files listed in meta, do not delete. This could be a bug!!
+            if (filesInMeta.isEmpty()) {
+                logger.warn("Meta file was empty. This should not happen. Getting out!!");
+                return;
+            }
+
+            // Delete the  old META files. We are giving start date which is so back in past to get
+            // all the META files.
+            // This feature did not exist in Jan 2018.
+            metas =
+                    metaProxy.findMetaFiles(
+                            new DateUtil.DateRange(
+                                    start_of_feature, dateToTtl.minus(1, ChronoUnit.HOURS)));
+
+            if (metas != null) {
+                logger.info(
+                        "Will delete(TTL) {} META files starting from: [{}]",
+                        metas.size(),
+                        metas.get(metas.size() - 1).getLastModified());
+                for (AbstractBackupPath meta : metas) {
+                    deleteFile(meta, false);
+                }
+            }
+
             Iterator<String> remoteFileLocations =
                     fileSystem.listFileSystem(getSSTPrefix(), null, null);
 
+            dateToTtl = dateToTtl.minus(config.getGracePeriodDaysForCompaction(), ChronoUnit.DAYS);
```

Comment:
Context on why you are considering `getGracePeriodDaysForCompaction` instead of TTL only.